nvdaHelper/local/nvdaControllerInternal.c

```diff
@@ -19,6 +19,11 @@ error_status_t __stdcall nvdaControllerInternal_requestRegistration(const wchar_
 	return _nvdaControllerInternal_requestRegistration(uuidString);
 }
 
+error_status_t(__stdcall *_nvdaControllerInternal_reportLiveRegion)(const wchar_t*, const wchar_t*);

```

Comment:
I do not understand this or the next couple of lines. what is this header for? How does this ultimately trigger the callback?

Reply:
These are stubs in the dll that rpc looks for and remotely calls within NVDA's process.  within each stub, the code calls a variable function pointer. These variable function pointers are assigned to point to a ctypes-wrapped Python function within NvDA's nvdaHelper.py. In short, this allows NVDA to provide Python implementations for these rpc functions.

Reply:
Look at nvdaHelper.py to see how these function pointers are assigned for the other rpc functions. Of course, the function pointer also needs to be exported from the dll in nvdaHelperLocal.def if not already.

Reply:
Ah, wait, NVDAHelper does patch this to start with an _ that's what I was missing.